File: docs/src/testers/sessions/lifecycle/error_handling.ts
// @sniptest filename=error_handling.ts
// @sniptest show=7-22
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();
let sessionId: string | null = null;
let errorMessage: string | undefined;

const session = client.Session();
const expectedError = new Error('Example automation failure');
try {
  await session.use(async () => {
    sessionId = session.getId();
    const status = await session.status();
    const page = await session.page();
    await page.goto('https://example.com');
    throw expectedError;
  });
} catch (error) {
  if (error !== expectedError) throw error;
  errorMessage = expectedError.message;
  console.log(`Automation failed: ${errorMessage}`);
}
// Cleanup happens even when automation throws.

export { sessionId as session_id, errorMessage as error_message };

File: docs/src/testers/sessions/lifecycle/error_handling.py
# @sniptest filename=error_handling.py
# @sniptest show=7-18
from notte_sdk import NotteClient

client = NotteClient()
session_id = None
error_message = None

session = client.Session(idle_timeout_minutes=2)
try:
    with session:
        session_id = session.session_id
        session.page.goto("https://example.com")
        raise ValueError("Example automation failure")
except ValueError as error:
    if str(error) != "Example automation failure":
        raise
    error_message = str(error)
    print(f"Automation failed: {error_message}")
# Cleanup happens even when automation raises.
